crates/lib3h/src/gateway/gateway_transport.rs

```diff
@@ -0,0 +1,157 @@
+#![allow(non_snake_case)]
+
+use crate::{
+    dht::{dht_protocol::*, dht_trait::Dht},
+    engine::p2p_protocol::P2pProtocol,
+    gateway::P2pGateway,
+    transport::{
+        error::{TransportError, TransportResult},
+        protocol::{TransportCommand, TransportEvent},
+        transport_trait::Transport,
+        TransportId, TransportIdRef,
+    },
+};
+use lib3h_protocol::DidWork;
+use rmp_serde::{Deserializer, Serializer};
+use serde::{Deserialize, Serialize};
+
+/// Compose Transport
+impl<T: Transport, D: Dht> Transport for P2pGateway<T, D> {
+    fn connect(&mut self, uri: &str) -> TransportResult<TransportId> {
+        self.inner_transport.borrow_mut().connect(&uri)
```

Comment:
Here we should be receiving the higher level identifier (encoded as a uri (`hc:HcSyada`?)). We need to do a peer discovery here to find the lower-level uri for connecting with our `inner_transport`.